src/content/basics/ha-masters/index.md

```diff
@@ -0,0 +1,169 @@
+worker [node pools](/basic/nodepools/) is taken into account.
+
+## Upgrades from previous releases {#upgrades}
+
+When upgrading a cluster to release v11.4.0, the cluster will remain a single
+master cluster during and after the upgrade. The API unavailability during the
+upgrade that is typical for single master clusters will apply for this upgrade.
+
+A conversion to high availability can be triggered manually after the upgrade
+is finished.
+
+## Conversion from single master to high availability {#conversion-to-ha}
+
+Single master clusters using release v{{% first_aws_ha_masters_version %}} or
+above on AWS can be converted to master node high availability in the user
+interfaces and via the APIs.
+
+## Via the web UI {#web-ui}
+
+The web UI presents information on the master node in the cluster details page.
+Next to these details you find a button _Switch to high availability…_, unless
+the cluster is currently undergoing an upgrade. Click this button and follow
+the instructions in the web UI.
+
+## Via the CLI (`gsctl`) {#gsctl}
+
+The `gsctl` CLI provides the [gsctl update cluster](/reference/gsctl/update-cluster/) to change cluster details. Check the reference for the `--master-ha` flag.
+
+## Via the Rest API {#rest-api}
+
+Check the [v5 cluster modification API reference](/api/#operation/modifyClusterV5) to find out how to convert a cluster programmatically using the Rest API.
+
+## Via the Control Plane K8s API {#cp-k8s-api}
+
+Two resources have to be changed in order to convert a single master cluster
+to high availability.
+
+1. `AWSControlPlane`
+2. `G8sControlPlane`
+
+### `AWSControlPlane`
+
+The cluster's [`AWSControlPlane`](/reference/cp-k8s-api/awscontrolplanes.infrastructure.giantswarm.io/)
+has a `.spec.availabilityZones` array which, in case of a single master
+cluster, has only one item. This item is the availability zone assigned to the
+only master node.
+
+To convert the cluster to high availability, this array has to be extended
+to have three items. Each item specifies the availability zone for one master
+node after the conversion.
+
+**CAUTION:** Order is important here. The first item must be left untouched. Or, in other words, what was the only array item before must be the first of three item after the change.
+
+Example for a cluster in region `eu-central-1`:
+
+The `AWSControlPlane` shows this spec (shortened):
+
+```yaml
+spec:
+  availabilityZones:
+  - eu-central-1c
+```
+
+The resource gets changed to this:
+
+```yaml
+spec:
+  availabilityZones:
+  - eu-central-1c
+  - eu-central-1a
+  - eu-central-1b
+```
+
+### `G8sControlPlane`
+
+The [`G8sControlPlane`](/reference/cp-k8s-api/g8scontrolplanes.infrastructure.giantswarm.io/)
+has a `.spec.replicas` attribute which indicates the number of master nodes to
+run. This is `1` for a single master cluster and needs to be modified to `3`.
+
+**Note:** Modifying the `G8sControlPlane` resource must happen _after_ the `AWSControlPlane`
```

Comment:
@anvddriesch @paurosello same here. Is the order important still?

Reply:
It is my understanding of this thread: https://gigantic.slack.com/archives/C0148R1G12N/p1592236645153400